flight/goto/mission/main.py

```diff
@@ -0,0 +1,74 @@
+"""
+Main driver code for moving drone to each waypoint
+"""
+
+import asyncio
+import json_parsing
+import goto
+from mavsdk import System
+import mavsdk as sdk
+import logging
+import math
+import typing
+from typing import Dict,List
+
+async def run() -> None:
+    #Put all latitudes, longitudes and altitudes into seperate arrays
+    lats: List[float]=[]
+    longs: List[float]=[]
+    altitudes: List[float]=[]
+    waypoints: List[Dict[str,float]] =json_parsing.waypoint_parsing("numbers.json")
+    for i in waypoints:
+        for key,val in i.items():
+            if(key=="latitude"):
+                lats.append(val)
+            if(key=="longitude"):
+                longs.append(val)
+            if(key=="altitude"):
+                altitudes.append(val)
+
+    #create a drone object
+    drone: System = System()
+    await drone.connect(system_address="udp://:14540")
+
+    #connect to the drone
+    print("Waiting for drone to connect...")
+    async for state in drone.core.connection_state():
+        if state.is_connected:
+            print("Drone discovered!")
+            break
+
+    print("Waiting for drone to have a global position estimate...")
+    async for health in drone.telemetry.health():
+        if health.is_global_position_ok:
+            print("Global position estimate ok")
+            break
+
+    print("-- Arming")
+    await drone.action.arm()
+
+    print("-- Taking off")
+    await drone.action.takeoff()
+
+    #wait for drone to take off
+    await asyncio.sleep(20)
+
+    #move to each waypoint in mission
+    for point in range(len(waypoints)):
+        await goto.move_to(drone,lats[point],longs[point],altitudes[point])
+
+    #return home
+    print("Last waypoint reached")
+    print("Returning to home")
+    await drone.action.return_to_launch()
+    print("Staying connected, press Ctrl-C to exit")
```

Comment:
You may want to add an exception handler for this escape; the KeyboardInterrupt is the exception name to use. The handler could force the drone to enter land mode, just to ease the decent when Ctrl+C is pressed.